src/plugins/data/public/ui/saved_query_form/helpers.tsx

```diff
@@ -0,0 +1,312 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useState, useCallback } from 'react';
+import {
+  EuiSmallButtonEmpty,
+  EuiSmallButton,
+  EuiForm,
+  EuiCompressedFormRow,
+  EuiCompressedFieldText,
+  EuiCompressedSwitch,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { sortBy, isEqual } from 'lodash';
+import { SavedQuery, SavedQueryService } from '../..';
+import { SavedQueryAttributes } from '../../query';
+
+interface Props {
+  savedQuery?: SavedQueryAttributes;
+  savedQueryService: SavedQueryService;
+  onSave: (savedQueryMeta: SavedQueryMeta) => void;
+  onClose: () => void;
+  formUiType: 'Modal' | 'Flyout';
+  showFilterOption: boolean | undefined;
+  showTimeFilterOption: boolean | undefined;
+  onSaveAsNew?: () => void;
+}
+
+export interface SavedQueryMeta {
+  title: string;
+  description: string;
+  shouldIncludeFilters: boolean;
+  shouldIncludeTimefilter: boolean;
+}
+
+export function useSaveQueryFormContent({
+  savedQuery,
+  savedQueryService,
+  onSave,
+  onClose,
+  showFilterOption = true,
+  showTimeFilterOption = true,
+  formUiType,
+  onSaveAsNew,
+}: Props) {
+  const [title, setTitle] = useState('');
+  const [enabledSaveButton, setEnabledSaveButton] = useState(false);
+  const [description, setDescription] = useState('');
+  const [savedQueries, setSavedQueries] = useState<SavedQuery[]>([]);
+  const [shouldIncludeFilters, setShouldIncludeFilters] = useState(true);
+  // Defaults to false because saved queries are meant to be as portable as possible and loading
+  // a saved query with a time filter will override whatever the current value of the global timepicker
+  // is. We expect this option to be used rarely and only when the user knows they want this behavior.
+  const [shouldIncludeTimefilter, setIncludeTimefilter] = useState(false);
+  const [formErrors, setFormErrors] = useState<string[]>([]);
+
+  // Instead of setting the default values based on the saved query above,
+  // we use this effect so that in case when user select "Save as new" in the flyout,
+  // the initial state resets with change to 'savedQuery'
+  useEffect(() => {
+    setTitle(savedQuery ? savedQuery.title : '');
```

Comment:
Wouldn't it be safer to do this?
```suggestion
    setTitle(savedQuery?.title || '');
```